metastore/src/test/scala/com/gu/tableversions/metastore/MetastoreSpec.scala

```diff
@@ -0,0 +1,175 @@
+package com.gu.tableversions.metastore
+
+import cats.effect.IO
+import com.gu.tableversions.core.Partition.PartitionColumn
+import com.gu.tableversions.core._
+import com.gu.tableversions.metastore.Metastore.TableChanges
+import com.gu.tableversions.metastore.Metastore.TableOperation._
+import org.scalatest.{FlatSpec, Matchers}
+
+/**
+  * Spec containing tests that apply across all Metastore implementations.
+  *
+  * These are black box tests purely in terms of the Metastore interface.
+  */
+trait MetastoreSpec {
+  this: FlatSpec with Matchers =>
+
+  def metastoreWithSnapshotSupport(
+      emptyMetastore: IO[Metastore[IO]],
+      initHiveTable: IO[Unit],
+      table: TableDefinition): Unit = {
+
+    it should "allow table versions to be updated for snapshot tables" in {
+
+      val scenario = for {
+        metastore <- emptyMetastore
+        _ <- initHiveTable
+
+        initialVersion <- metastore.currentVersion(table.name)
+
+        _ <- metastore.update(table.name, TableChanges(List(UpdateTableLocation(table.location, VersionNumber(1)))))
```

Comment:
I don't think the version number should be manipulated explicitely by the end user. However do we want to abstract it within `Metastore` or at a higher level?

Reply:
See points above, but I don't think this interface is something for the "end user", it's a fairly low level service the makes an underlying metastore version aware (hmm, maybe `VersionAwareMetastore` or `VersionedMetastore` would be a better name for this).

In the real code, the versions passed into the metastore's methods would come from the version store (`TableVersions` right now) and not be magicked up by clients..

Reply:
I had a look at this but on reflection, I think it's better that this particular interface is explicit about the changes it will perform to tables in the underlying metastore. I don't think this should leak into the higher level interfaces anyway. Might see if I can write some docs around this to explain what this is the case.

Reply:
Make sense. So this `Metastore` API (or `VersionedMetastore` 😉) is low level and will be used by the user-facing layer (which I guess will take care of keeping track of version history)